### PR TITLE
readme: Add repository deprecation notice

### PR DESCRIPTION
**Pull request by chris-crone**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,6 +7,25 @@ version for the lifetime of the product.
...
 
+:warning: **This repository is now deprecated and will be archived** :warning:
+
+Starting with the Docker 20.10 release, packages for the Docker Engine and
+Docker CLI are built directly from their respective source repositories instead
+of from this repository.
+
+Practically this means:
+1. This repository is no longer the “source of truth” for Docker CE builds.
+1. The commit SHA and tag for Docker CLI build will come from the
```
Comment (thaJeztah) [inline README.md:18]: I tend to still number each (but know it's not needed for the rendered Markdown). No need to change if there's no other review comments though

**Review by thaJeztah — APPROVED** [review]
LGTM, thanks!

File: README.md (hunk 1 above)
Comment (chris-crone) [reply README.md:18]: Happy to change that :)

File: README.md [hunk 2]
```diff
@@ -7,6 +7,25 @@ version for the lifetime of the product.
 
 This repository is solely maintained by Docker, Inc.
 
+:warning: **This repository is now deprecated and will be archived** :warning:
```
Comment (tiborvass) [inline README.md:10]: Maybe put it further up so it's the first thing we see?

**Comment by mark-choi** [comment]
Hi,
Does it mean no longer Docker-CE 19.03.xx will not exist?

**Comment by thaJeztah** [comment]
@mark-choi this deprecation is only about this mono-repo. This repository contains a copy of source code from several upstream repositories, which was mostly done to have all code in a single place for easier building. For example, for the 19.03 release, this repository has:

- a copy of the upstream https://github.com/docker/docker/tree/19.03 release branch (copied in https://github.com/docker/docker-ce/tree/19.03/components/engine)
- a copy of the upstream https://github.com/docker/docker/cli/19.03 release branch (copied in https://github.com/docker/docker-ce/tree/19.03/components/cli)
- a copy of the upstream https://github.com/docker/docker-ce-packaging/tree/19.03 release branch (copied in https://github.com/docker/docker-ce/tree/19.03/components/packaging)


At the time when this repository was created all those went into a single `.deb` or `.rpm` package. However, current versions of  Docker (including 19.03) now have separate packages for the "engine" (built from https://github.com/docker/docker), the "cli" (built from https://github.com/docker/cli), as well as other binaries (e.g. https://github.com/docker/buildx). As all of those are developed separately, and potentially with different release-cycles, this repository is now actually making things more _complicated_, not _easier_.

For that reason, we're deprecating it;

- we'll continue updating the _current_ (19.03) release branch until 19.03 reaches EOL
- but we won't be creating a 20.10 release branch

Once 19.03 reaches EOL, this repository will be archived ("read only"), and moved to our "docker-archive" organization on GitHub, which is where we keep repositories that are no longer maintained; https://github.com/docker-archive.
When moving repositories, GitHub adds an automatic redirect to the new location though, so scripts that still want to use one of the release branches from this repository should still work and find the new location.

**Comment by thaJeztah** [comment]
Thinking of which; perhaps we should at some point empty the default (`master`) branch in this repository to "forcibly" break users that use the master branch (not a release branch), because the master branch will no longer be updated, and we don't want users to be using outdated source-code (potentially with unpatched vulnerabilities) without being aware.

File: README.md (hunk 2 above)
Comment (chris-crone) [reply README.md:10]: Done!

File: README.md [hunk 3]
```diff
@@ -1,5 +1,28 @@
...
+   [docker/cli](https://github.com/docker/cli) repository and the commit SHA and
+   tag for the Docker Engine will come from the
+   [moby/moby](https://github.com/moby/moby) repository.
+3. Release branches for the Engine, CLI, and packaging will be maintained on
+   their respective repositories.
+4. Updates will stop being made to this repository and it will be archived in
+   the future.
+5. Changelog is now
+   [Release Notes](https://docs.docker.com/engine/release-notes/).
+6. The `master` branch of this repository will be emptied when the repository
```
Comment (chris-crone) [inline README.md:21]: @thaJeztah I added this at your suggestion in the comments.

**Review by justincormack — APPROVED** [review]

File: README.md (hunk 3 above)
Comment (thaJeztah) [reply README.md:21]: Yes, looks good. We can update the readme to link to the release-branches (either when we clean it, or before that); not a stopper for this PR though.

Comment (mark-choi) [reply README.md:21]: Hi Chris,

Many thanks !!!

**Pull request merged**